app/gui2/src/components/DocumentationPanel.vue

```diff
@@ -0,0 +1,277 @@
+<script setup lang="ts">
+import { default as DocsExamples } from '@/components/documentation/DocsExamples.vue'
+import { default as DocsHeader } from '@/components/documentation/DocsHeader.vue'
+import { default as DocsList } from '@/components/documentation/DocsList.vue'
+import { default as DocsSynopsis } from '@/components/documentation/DocsSynopsis.vue'
+import { default as DocsTags } from '@/components/documentation/DocsTags.vue'
+import { useSuggestionDbStore, type SuggestionDb } from '@/stores/suggestionDatabase'
+import type { SuggestionEntry, SuggestionId } from '@/stores/suggestionDatabase/entry'
+import { SuggestionKind } from '@/stores/suggestionDatabase/entry'
+import type { Doc } from '@/util/docParser'
+import { type Opt } from '@/util/opt'
+import type { SuggestionEntryArgument } from 'shared/languageServerTypes/suggestions'
+import { computed } from 'vue'
+
+const props = defineProps<{ selectedEntry: Opt<SuggestionId> }>()
+const emit = defineEmits<{ 'update:selectedEntry': [id: SuggestionId] }>()
+
+// === Types ===
+
+/**
+ * Intermediate representation of the entries documentation.
+ */
+export type Docs =
+  | { Function: FunctionDocs }
+  | { Type: TypeDocs }
+  | { Module: ModuleDocs }
+  | { Local: LocalDocs }
+  | { Placeholder: string }
```

Comment:
This whole type and corresponding functions should be moved out of the component, possibly to a `docParser` module. Exporting types from components is rarely a good idea.

Also, If I understand this correctly, all objects of `Docs` type are constructed manually in code (i.e. not from parser), so there is no point in using this unconventional type union format. It is better to use tagged unions that typescripts understands well. One nice way to do it is to [include a common "kind" property in all union interfaces](https://mariusschulz.com/blog/tagged-union-types-in-typescript#modeling-payment-methods-with-tagged-union-types). You could even make an `enum` for the available kinds, similarily to `SuggestionKind`, but it's not strictly necesasry. In general, that should remove a lot of type gymnastics that you had to do in the following code.

Reply:
Types and supporting functions were moved to a separate module using the suggested approach. The code indeed became much more straightforward!